module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,57 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+
+    for (const auto& arr : bitmap) {
+        for (const auto& el : arr) {
+            if (el <= ' ')
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& grayScale) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width);
+    int count = 1, tmp;
+
+    for (auto row = grayScale.begin(); row != grayScale.end(); row++) {
+        for (auto column = row->begin(); column != std::prev(row->end()); column++) {
+            if (*(std::next(column)) != *column) {
+                compressed.push_back(std::make_pair(*column, count));
+                tmp = *(std::next(column));
+                count = 1;
+                continue;
+            }
+            count++;
+            tmp = *column;
+        }
+        compressed.push_back(std::make_pair(tmp, count));
+        count = 1;
```

Comment:
2 times you push back {color, counter} maybe it could be done once? Try other construction of loop.